presto-main/src/main/java/com/facebook/presto/operator/HashAggregationOperator.java

```diff
@@ -286,6 +286,14 @@ public void addInput(Page page)
     }
 
     @Override
+    public void revokeMemory()
+    {
+        if (aggregationBuilder != null && !aggregationBuilder.isBusy()) {
```

Comment:
move the `!aggregationBuilder.isBusy()` check to inside `aBuilder.revokeMemory()`. It is not obvious in this context that `isBusy` means `isSpillingAlready`.